contracts/Oracle/Calculations/Balancer.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.12;
+import "../../Utilities/Ownable.sol";
+
+interface IERC20 {
+    function decimals() external view returns (uint8);
+    function symbol() external view returns (string memory);
+    function balanceOf(address) external view returns (uint256);
+}
+
+interface IBalancerVault {
+    function getPoolTokens(bytes32 poolId)
+        external
+        view
+        returns (
+            IERC20[] calldata tokens,
+            uint256[] calldata balances,
+            uint256 lastChangeBlock
+        );
+}
+
+interface IBalancerPool is IERC20 {
+    function getPoolId() external view returns (bytes32 poolId);
+    function totalSupply() external view returns (uint256);
+    function getActualSupply() external view returns (uint256);
+    function getVirtualSupply() external view returns (uint256);
+}
+
+interface IWrappedAsset {
+    function ASSET() external view returns (address);
+    function ATOKEN() external view returns (address);
+}
+
+interface IOracle {
+    function getPriceUsdcRecommended(address tokenAddress)
+        external
+        view
+        returns (uint256);
+    function usdcAddress() external view returns (address);
+}
+
+interface IYearnAddressesProvider {
+    function addressById(string memory) external view returns (address);
+}
+
+contract CalculationsBalancer is Ownable {
+    IYearnAddressesProvider internal yearnAddressesProvider;
+
+    IBalancerVault internal constant balancerVault =
+        IBalancerVault(0xBA12222222228d8Ba445958a75a0704d566BF2C8);
+
+    constructor(address _yearnAddressesProviderAddress) {
+        yearnAddressesProvider = IYearnAddressesProvider(
+            _yearnAddressesProviderAddress
+        );
+    }
+
+    function updateYearnAddressesProviderAddress(
+        address _yearnAddressesProviderAddress
+    ) external onlyOwner {
+        yearnAddressesProvider = IYearnAddressesProvider(
+            _yearnAddressesProviderAddress
+        );
+    }
+
+    function oracle() internal view returns (IOracle) {
+        return IOracle(yearnAddressesProvider.addressById("ORACLE"));
+    }
+
+    //Phantom pools
+    //Stable Pools
+    function poolLpTotalValueUsdc(address lpAddress)
+        public
+        view
+        returns (uint256)
+    {
+        //If the token is a Wrapped Aave asset return the total atoken value held
+        try IWrappedAsset(lpAddress).ASSET() returns (address _asset) {
+            IERC20 aToken = IERC20(IWrappedAsset(lpAddress).ATOKEN());
+            return
+                (oracle().getPriceUsdcRecommended(_asset) *
+                    aToken.balanceOf(lpAddress)) / (10**aToken.decimals());
+        } catch {}
+
+        bytes32 poolId = IBalancerPool(lpAddress).getPoolId();
+
+        (IERC20[] memory tokens, uint256[] memory balances, ) = balancerVault
+            .getPoolTokens(poolId);
+        uint256 totalValue;
+        for (uint256 i; i < tokens.length; i++) {
```

Comment:
Cheaper gas-wise to do ++i instead of i++

Reply:
added, and a couple other small optimizations https://github.com/yearn/yearn-lens/pull/57/commits/688e7488fcc72184543b70fb231fb37bb73189db